processor/src/debug.rs

```diff
@@ -28,15 +28,21 @@ impl fmt::Display for VmState {
 /// at each clock cycle.
 /// If the execution returned an error, it returns that error on the clock cycle
 /// it stopped.
-pub struct VmStateIterator {
-    process: Process,
+pub struct VmStateIterator<A>
+where
+    A: AdviceProvider,
+{
+    process: Process<A>,
```

Comment:
I actually wonder if we need to make this generic. The only reason for it to be generic is that we currently store the full process here. But we actually don't need access to advice provider at all. So, maybe we should break the process into the components, and store only the ones that we actually need here. These are: `system`, `decoder`, `stack`, and `chiplets`. We could store them in individual fields or create a wrapper struct of some sort.

Reply:
I think it's better to create a wrapper struct as we would have that very same struct on `Process`. I'll just review a couple of things regarding these two structs and get back to this topic

Reply:
I would actually not modify `Process` struct further. I was thinking that the wrapper struct would be in `VmStateIterator`- and even there, I'm not sure it is really necessary as we only need 4 elements and we don't really rely on any `Process` level methods.

Reply:
A wrapper struct would be good only if it can be used in multiple places. Alternatively, we can have an `into_parts` method of `Process` that will expose its attributes - allowing us to build the iterator from that. wdyt?

Reply:
We already do have something like `into_parts()` - but it is called [to_components()](https://github.com/0xPolygonMiden/miden-vm/blob/next/processor/src/lib.rs#L406). `into_parts()` would probably have been a more canonical name though.